testtracer_test.go

```diff
@@ -39,10 +46,39 @@ func (n testTracer) StartSpanWithOptions(opts StartSpanOptions) Span {
 
 // Inject belongs to the Tracer interface.
 func (n testTracer) Inject(sp Span, format interface{}, carrier interface{}) error {
-	return nil
+	span := sp.(testSpan)
+	switch format {
+	case TextMap:
+		// Just for testing purposes... generally not a worthwhile thing to
+		// propagate.
+		carrier.(TextMapWriter).Add(testHTTPHeaderPrefix+"opname", span.OperationName)
+		carrier.(TextMapWriter).Add(testHTTPHeaderPrefix+"hasparent", strconv.FormatBool(span.HasParent))
```

Comment:
it feels like we're setting a bad example by adding opname and hasparent as baggage.